torch/distributed/fsdp/_init_utils.py

```diff
@@ -586,42 +592,27 @@ def _get_ignored_modules(
 def _get_ignored_params(
     root_module: torch.nn.Module,
     ignored_modules: Set[torch.nn.Module],
-) -> Tuple[Set[torch.nn.Parameter], Set[str]]:
+    ignored_parameters: Optional[Iterable[torch.nn.Parameter]],
+) -> Set[torch.nn.Parameter]:
     """
-    Returns the parameters of the modules in ``ignored_modules``,
-    excluding any :class:`FlatParameter` s, and their fully prefixed names,
-    both as :class:`set` s.
+    Returns the parameters of the modules in ``ignored_modules`` and 
+    the parameters in ``ignored_parameters``, excluding any :class:`FlatParameter` s.
     """
-    ignored_params_to_names = dict(
-        (p, n)
+    all_ignored_params = set()
+    params_in_ignored_modules = list(
+        p
         for m in ignored_modules
-        for n, p in m.named_parameters()
+        for p in m.parameters()
         if not _is_fsdp_flattened(p)
     )
-    # Conservatively include all shared parameters' names
-    param_to_unflat_param_names = _get_param_to_fqns(
-        root_module,
-        dedup_shared_params=False,
+    params_in_ignored_parameters = list(
+        p
+        for p in ignored_parameters
+        if not _is_fsdp_flattened(p)
```

Comment:
is it possible to pass an FSDP flattened ignored parameter here and have it indeed be filtered out? Can we add a test that would simulate this?

Reply:
yes, the unit tests have already covered this case